src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/PooledDisposer.cs

```diff
@@ -8,9 +8,7 @@
 namespace Microsoft.CodeAnalysis.PooledObjects;
 
 [NonCopyable]
```

Comment:
Adds `[NonCopyable]` attribute to the type definition.

Reply:
What about just requiring the consumer to provide this attribute in all cases?

Reply:
Isn't the analyzer Roslyn specific? I would rather not impose more requirements on consumers of this source package.

That said, I'm trying to put together another source package (say `Microsoft.CodeAnalysis.Contracts`) with basic contracts and exception utilities, which are used by our threading utilities (extracted to source package https://github.com/dotnet/roslyn/pull/76937).

We can include NonCopyable there. It's a contract after all, so it would fit.

Reply:
https://github.com/dotnet/roslyn/pull/76997